crates/router/src/db/mandate.rs

```diff
@@ -108,46 +111,149 @@ impl MandateInterface for Store {
 impl MandateInterface for MockDb {
     async fn find_mandate_by_merchant_id_mandate_id(
         &self,
-        _merchant_id: &str,
-        _mandate_id: &str,
+        merchant_id: &str,
+        mandate_id: &str,
     ) -> CustomResult<storage::Mandate, errors::StorageError> {
-        // [#172]: Implement function for `MockDb`
-        Err(errors::StorageError::MockDbError)?
+        match self
+            .mandates
+            .lock()
+            .await
+            .iter()
+            .find(|mandate| mandate.merchant_id == merchant_id && mandate.mandate_id == mandate_id)
+        {
+            Some(mandate) => Ok(mandate.clone()),
+            None => {
+                Err(errors::StorageError::ValueNotFound("mandate not found".to_string()).into())
+            }
+        }
```

Comment:
instead of match, can we go with a more functional approach, not a hard requirement, but as a improvement, we can consider using `.ok_or_else` after `.find`